testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -178,6 +179,55 @@ private String describeMemoryPool(TestingTrinoServer coordinator, TestingTrinoSe
         return result.toString();
     }
 
+    private void checkQueryInfosFinal()
+    {
+        tryGetDistributedQueryRunner().ifPresent(runner -> assertEventually(
+                new Duration(30, SECONDS),
+                new Duration(1, SECONDS),
+                () -> {
+                    TestingTrinoServer coordinator = runner.getCoordinator();
+                    QueryManager queryManager = coordinator.getQueryManager();
+                    for (BasicQueryInfo basicQueryInfo : queryManager.getQueries()) {
+                        QueryId queryId = basicQueryInfo.getQueryId();
+                        if (!basicQueryInfo.getState().isDone()) {
+                            fail("query is expected to be in done state: " + basicQueryInfo.getQuery());
+                        }
+                        QueryInfo queryInfo = queryManager.getFullQueryInfo(queryId);
+                        if (!queryInfo.isFinalQueryInfo()) {
+                            fail("QueryInfo is expected to be final: " + basicQueryInfo.getQuery());
+                        }
+                    }
+                }));
+    }
+
+    private void checkTasksDone()
+    {
+        tryGetDistributedQueryRunner().ifPresent(runner -> assertEventually(
+                new Duration(30, SECONDS),
+                new Duration(1, SECONDS),
+                () -> {
+                    QueryManager queryManager = runner.getCoordinator().getQueryManager();
+                    List<TestingTrinoServer> servers = runner.getServers();
+                    for (TestingTrinoServer server : servers) {
+                        SqlTaskManager taskManager = server.getTaskManager();
+                        List<TaskInfo> taskInfos = taskManager.getAllTaskInfo();
+                        for (TaskInfo taskInfo : taskInfos) {
+                            TaskId taskId = taskInfo.getTaskStatus().getTaskId();
+                            QueryId queryId = taskId.getQueryId();
+                            String query = "unknown";
+                            try {
+                                query = queryManager.getQueryInfo(queryId).getQuery();
+                            }
+                            catch (NoSuchElementException ignored) {
+                            }
```

Comment:
nit: move to previous line

Reply:
When I move it to the previous line it doesn't pass the style check

Reply:
Interesting. I have checkstyle set up in IntelliJ and it did not complain. 🤷

Reply:
Yeah, my Intellij doesn't complain either (though it is trying to format it with a new line when I try to autoformat). But then the check during the build fails.